plugins/pulp_rpm/plugins/db/models.py

```diff
@@ -1005,6 +1011,33 @@ def verify_size(self, location):
             with open(location) as fp:
                 verification.verify_size(fp, self.size)
 
+    def set_repodata(self, metadata_type, xml_snippet):
+        """
+        Compress metadata and put it into `repodata` attribute which will be saved to the db later.
+
+        :param metadata_type: key for the `repodata` dictionary which indicates type of metadata
+        :type  metadata_type: str
+        :param xml_snippet: utf-8 string which will be compressed and put into `repodata` dictionary
+        :type  xml_snippet: str
+        """
+        if metadata_type in constants.MANDATORY_METADATA_TYPES:
+            if isinstance(xml_snippet, unicode):
+                xml_snippet = xml_snippet.encode('utf-8')
+            self.repodata[metadata_type] = bson.binary.Binary(gzip.zlib.compress(xml_snippet))
+
+    def get_repodata(self, metadata_type):
```

Comment:
@mhrivnak , here is some migration test results:
* 1.5ms per unit (50s for 35K packages) on my dev VM which does almost nothing except running this migration
* 4ms per unit (15min for 220K packages) on a real-world customer busy system
* 4ms per unit (5.5min for 81.5K packages) on a customer-like busy system

With batch_size == 100 -> no performance degradation observed.
With batch_size == 10 -> 9% of performance degradation observed but I do not think there is a need for such a small batch_size in our case.

It's good that on both real systems results are quite consistent but maybe it's not enough to be confident ;)